cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java

```diff
@@ -259,14 +261,30 @@ protected void onSaveInstanceState(Bundle outState) {
     mIntentRequestTracker.saveInstanceState(outState);
   }
 
-  // Initially copied from ContentShellActiviy.java
+  @Override
+  public boolean onKeyDown(int keyCode, KeyEvent event) {
+    if (keyCode == KeyEvent.KEYCODE_BACK) {
+      WebContents webContents = getActiveWebContents();
+      if (webContents != null) {
```

Comment:
In general I'd advocate for early return instead of multiple nested conditions, i.e.
```
if (keyCode != KeyEvent.KEYCODE_BACK) {
  return super.onKeyDown(keyCode, event);
}
WebContents webContents = getActiveWebContents();
if (webContents == null) {
  return super.onKeyDown(keyCode, event);
}
// etc
```

Same for onKeyUp() below.

Also can |webContents| in l.267 be null?

Reply:
I guess what confused me is that `getActiveWebContents()` ([1]) is not annotated @Nullable ([2]) but on inspection it clearly should be .

[1] https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java;l=343?q=getActiveWebContents&sq=&ss=h%2Flbshell-internal%2Fcobalt_src%2F%2B%2Frefs%2Fheads%2Fmain:cobalt%2Fandroid%2Fapk%2Fapp%2Fsrc%2Fmain%2Fjava%2Fdev%2Fcobalt%2Fcoat%2F

[2] https://google.github.io/styleguide/javaguide.html#s4.8.5-annotations